qa-pipelines/tasks/lib/azure-aks.sh

```diff
@@ -2,29 +2,25 @@
 
 set -eu # errexit, nounset
 
-az_login_error_msg() {
-    cat << 'EOF' >& 2
-    kubectl create secret generic -n kube-system aks-dns-sp \
-        --from-literal appId=${AZ_SP_APPID} \
-        --from-literal tenant=${AZ_SP_TENANT} \
-        --from-literal password=${AZ_SP_PASSWORD}
-EOF
+# Outputs a json representation of a yml document. For multi-document files, only the first document will be converted
+y2j() {
+    ruby -r json/ext -r yaml -e "puts YAML.load_file('$1').to_json"
 }
 
 az_login() {
-    local sp_app_id sp_password sp_tenant
-    sp_app_id=$(kubectl get secret -n kube-system -o json aks-dns-sp | jq -r .data.appId | base64 -d)
-    sp_password=$(kubectl get secret -n kube-system -o json aks-dns-sp | jq -r .data.password | base64 -d)
-    sp_tenant=$(kubectl get secret -n kube-system -o json aks-dns-sp | jq -r .data.tenant | base64 -d)
-    if ! az login --service-principal -u ${sp_app_id} --password ${sp_password} --tenant ${sp_tenant}; then
-        az_login_error_msg
+    if ! az login --service-principal -u ${AZ_SP_APPID} --password ${AZ_SP_PASSWORD} --tenant ${AZ_SP_TENANT}; then
+        echo "Azure login error"
         exit 1
     fi
 }
 
 AZURE_DNS_ZONE_NAME=susecap.net
 AZURE_DNS_RESOURCE_GROUP=susecap-domain
-AZURE_AKS_RESOURCE_GROUP=$(kubectl get configmap -n kube-system -o json cap-values | jq -r '.data["resource-group"]')
+if grep "eks.amazonaws.com" ~/.kube/config; then
+    AZURE_AKS_RESOURCE_GROUP=$(y2j ~/.kube/config clusters | jq -r .clusters[0].name | cut -d / -f 2)
```

Comment:
This second param to y2j doesn't get used right?

Reply:
anh, good catch!